src/components/tabBar/index.tsx

```diff
@@ -172,154 +195,103 @@ class TabBar extends Component<TabBarProps, State> {
-          showsHorizontalScrollIndicator={false}
-          scrollEnabled={scrollEnabled}
-          scrollEventThrottle={16}
-          onScroll={this.onScroll}
-          onContentSizeChange={this.onContentSizeChange}
-          contentContainerStyle={{minWidth: '100%'}}
-        >
-          <View row style={this.styles.tabBar}>
-            {this.renderChildren()}
-          </View>
-        </ScrollView>
-        {this.renderGradient(containerHeight, backgroundColor)}
-      </View>
+      <ScrollBar
+        ref={this.scrollBar}
+        contentContainerStyle={{minWidth: '100%'}}
+        scrollEnabled={scrollEnabled}
+        scrollEventThrottle={16}
+        onScroll={this.onScroll}
+        onContentSizeChange={this.onContentSizeChange}
+        height={containerHeight}
+        gradientColor={backgroundColor}
+        containerView={containerView}
+        containerProps={containerProps}
+        gradientMargins={gradientMargins}
+      >
+        <View row style={this.styles.tabBar}>
+          {this.renderChildren()}
+        </View>
+      </ScrollBar>
     );
   }
 
-  shouldBeMarked = (index: number) => {
-    return this.state.currentIndex === index && !this.isIgnored(index) && this.childrenCount > 1;
-  };
-
   renderChildren() {
     this.itemsRefs = [];
-    const {indicatorStyle} = this.props;
-
-    const children = React.Children.map(this.props.children, (child, index) => {
-      if (React.isValidElement(child)) {
-        return React.cloneElement(child, {
-          indicatorStyle,
-          selected: this.shouldBeMarked(index),
-          onPress: () => {
-            this.onItemPress(index, child.props);
-          },
-          ref: (r: any) => {
-            this.itemsRefs[index] = r;
-          }
-        });
-      }
+    const {indicatorStyle, darkTheme} = this.getThemeProps();
+
+    const children = React.Children.map<any, TabBarItem>(this.props.children as TabBarItem[], (child, index) => {
+      const accessLabel = child.props.accessibilityLabel || child.props.label || '';
+
+      //TODO: review it again, all types here should be correct. As from React.Children.map it gets definitely child: React.ReactNode, and React.cloneElement does not accept it.
+      // But seems it's work in a real life, so maybe it is just trouble with types compatibility
+      //@ts-ignore
+      return React.cloneElement(child, {
+        indicatorStyle,
+        darkTheme,
+        selected: this.shouldBeMarked(index),
+        onPress: () => {
+          this.onItemPress(index, child.props);
+        },
+        ref: r => {
+          this.itemsRefs[index] = r;
+        },
+        accessibilityLabel: `${accessLabel} ${index + 1} out of ${this.childrenCount}`
+      });
     });
     return children;
   }
 
   render() {
-    const {enableShadow, style} = this.props;
+    const {enableShadow, style, containerView, containerWidth} = this.getThemeProps();
+    const Container = containerView ? containerView : View;
```

Comment:
I assume containerView prop is for supporting the tablet margin? 
Have you considered doing it with style props? 
Similar to what we have in TabController.TabBar component
https://github.com/wix/react-native-ui-lib/blob/0be47258246b3ed3f69882e8e5a826bc3b5f4ebf/src/components/tabController/TabBar.js#L408

Reply:
I did consider it, but I need to wrap both the component and the ScrollBar. Also, I need a view that will take care of updates when orientation changes, which is what I intend to pass as a 'containerView' for both.